cmd/allowPrivilegeEscalation.go

```diff
@@ -48,10 +48,16 @@ func checkAllowPrivilegeEscalation(container Container, result *Result) {
 
 func auditAllowPrivilegeEscalation(resource k8sRuntime.Object) (results []Result) {
 	for _, container := range getContainers(resource) {
-		result := newResultFromResource(resource)
-		checkAllowPrivilegeEscalation(container, &result)
+		result, err := newResultFromResource(resource)
+		if err == ErrResourceTypeNotSupported {
```

Comment:
That code is weird .. ?

If we add new errors they will all be logged as "Resource type not supported" even if they say something else ..
the reason it is checked separately is because we want to skip unknown resource types (ie. if we get a Service, we have no validations to run, so silently skip it)

Reply:
@luna-duclos I agree if we want to extend that interface further for multiple errors then it should be something like:  
```
switch err {
    case errResourceTypeNotSupported:
}
```

but silently skipping is bit weird because then when we are never telling a user that the audit was skipped for a particular resource. So at least logging that error would be a good idea.

Reply:
@luna-duclos I did not get it like how exactly you will obtain the resource type which is not supported in `getResources` function ?